pyproject.toml

```diff
@@ -18,6 +18,7 @@ dependencies = [
     "networkx~=3.0",
     "typer[all]~=0.9.0",
     "xlsxwriter~=3.2.0",
+    "types-requests~=2.32.0.20240622",
```

Comment:
2 things here, you might want to add types request to the dev dependencies, and you'll need to include requests in the actual package dependencies:

```diff
diff --git a/pyproject.toml b/pyproject.toml
index 71abf04..845842a 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -18,7 +18,8 @@ dependencies = [
     "networkx~=3.0",
     "typer[all]~=0.9.0",
     "xlsxwriter~=3.2.0",
-    "types-requests~=2.32.0.20240622",
+    "requests~=2.32.3",
+    "pydantic~=2.8.2",
 ]
 
 [project.optional-dependencies]
@@ -27,6 +28,7 @@ dev = [
     "pre-commit~=2.20.0",
     "ruff~=0.3.3",
     "mypy~=1.9.0",
+    "types-requests~=2.32.0.20240622",
 ]
 
 [project.scripts]
```